aws/cloudformation/drone-stack.yml

```diff
@@ -0,0 +1,367 @@
+        InstanceType: t3.small
+        Monitoring:
+          Enabled: true
+        UserData: 
+          Fn::Base64:
+            !Sub 
+              - "#!/bin/bash\necho ECS_CLUSTER=${EcsClusterName} >> /etc/ecs/ecs.config;echo ECS_BACKEND_HOST= >> /etc/ecs/ecs.config;"
+              - {EcsClusterName: !Ref DroneMasterECSCluster}
+ 
+  DroneMasterEcsInstanceAsg:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      AvailabilityZones: !Ref AZs
+      LaunchTemplate:
+        LaunchTemplateId: !Ref DroneMasterLaunchTemplate
+        Version: !GetAtt DroneMasterLaunchTemplate.LatestVersionNumber
+      MinSize: 1
+      MaxSize: 1
+      DesiredCapacity: 1
+
+  DroneMasterCloudwatchLogsGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub "DroneMaster-${AWS::StackName}"
+      RetentionInDays: 14
+
+  DroneMasterTaskDefinition:
+    Type: AWS::ECS::TaskDefinition
+    Properties:
+      Family: !Sub "${AWS::StackName}-drone-master"
+      RequiresCompatibilities:
+        - EC2
+      ExecutionRoleArn: !Sub "arn:aws:iam::${AWS::AccountId}:role/DroneEcsTaskExecutionRole"
+      Volumes:
+        -
+          Name: db-data
+          Host:
+            SourcePath: /var/lib/drone
+      ContainerDefinitions:
+        - 
+          Name: drone-master
+          Essential: true
+          Image: drone/drone:1.0.0-rc.5
+          MemoryReservation: 512
+          LogConfiguration:
+            LogDriver: awslogs
+            Options:
+              awslogs-group: !Ref DroneMasterCloudwatchLogsGroup
+              awslogs-region: !Ref AWS::Region
+              awslogs-stream-prefix: ecs
+          PortMappings:
+            - 
+              ContainerPort: 80
+            -
+              ContainerPort: 443
+          MountPoints:
+            -
+              ContainerPath: /data
+              SourceVolume: db-data
+          Environment:
+            -
+              Name: DRONE_GITHUB_SERVER 
+              Value: https://github.com
+            -
+              Name: DRONE_SERVER_HOST 
+              Value: !GetAtt DroneMasterELB.DNSName
+            -
+              Name: DRONE_SERVER_PROTO
+              Value: http
+            -
+              Name: DRONE_TLS_AUTOCERT
+              Value: true
+            # ECS secrets manager integration not supported by Cloudformation yet: https://github.com/aws/containers-roadmap/issues/97
+            # Instead, we must use "dynamic references": https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html
+            -
+              Name: DRONE_GITHUB_CLIENT_ID
+              Value: '{{resolve:secretsmanager:drone/DRONE_GITHUB_CLIENT_ID}}'
+            -
+              Name: DRONE_GITHUB_CLIENT_SECRET
+              Value: '{{resolve:secretsmanager:drone/DRONE_GITHUB_CLIENT_SECRET}}'
+            -
+              Name: DRONE_RPC_SECRET
+              Value: '{{resolve:secretsmanager:drone/DRONE_RPC_SECRET}}'
```

Comment:
Could the Github-client credentials be combined into a single JSON secret, e.g.: `{{resolve:secretsmanager:drone/github:secret-string:CLIENT_ID}}` and `{{resolve:secretsmanager:drone/github:secret-string:CLIENT_SECRET}}`? (or maybe also pack the `DRONE_RPC_SECRET` value into the same Secret as well?)

Not super-important for this one case, just thinking ahead about long-term organization/grouping of credentials into secrets using SecretsManager.

Reply:
That's a good point, I should try to follow the best practices we'd talked about. I'll try to do one secret per "secured service", so it'll look like:

* drone/master
  * GITHUB_CLIENT_ID
  * GITHUB_CLIENT_SECRET
* drone/shared
  * DRONE_RPC_SECRET